src/pocketmine/Player.php

```diff
@@ -2685,7 +2659,23 @@ public function attack($damage, $source = EntityDamageEvent::CAUSE_MAGIC){
 		parent::attack($damage, $source);
 
 		if($source instanceof EntityDamageEvent and $source->isCancelled()){
```

Comment:
Can you commit a patch? You can edit that file on your branch using the GitHub web interface, it'll show up here.